lib/commands/connect/import.js

```diff
@@ -15,21 +17,14 @@ module.exports = {
     ],
     needsApp: true,
     needsAuth: true,
-    run: function (context) {
-      api.withUserConnections(context.auth.password, context.app, context.flags).then(function(connections){
-          if (connections.length > 1) {
-            console.log("Multiple connections found. Please use '-r' to specify a single connection by resource name.");
-          } else {
-            let connection = connections[0];
-            let url = 'https://' + api.HOST + ':443' + '/api/v3/connections/' + connection.id + '/actions/import';
-            fs.readFile(context.args.file, 'utf8', function(err, data) {
-              data = JSON.parse(data);
-              api.request(context.auth.password, 'POST', url, data).then(function(response) {
-                console.log("Upload complete");
-                console.log(response.text);
-              });
-            });
-          }
-      });
-    }
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connection = yield api.withConnection(context, heroku);
+      let url = '/api/v3/connections/' + connection.id + '/actions/import';
+      fs.readFile(context.args.file, 'utf8', co.wrap(function* (err, data) {
```

Comment:
`fs` functions don't use promises, so the `yield` trick doesn't seem to work here.